### PR TITLE
Api implementation

### PR DESCRIPTION
**Pull request by polina-moisenia**
HW2

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,5 +17,18 @@
   "bugs": {
     "url": "https://github.com/polina-moisenia/hello-blog/issues"
   },
-  "homepage": "https://github.com/polina-moisenia/hello-blog#readme"
+  "homepage": "https://github.com/polina-moisenia/hello-blog#readme",
+  "dependencies": {
+    "bcrypt": "^3.0.6",
+    "cookie-parser": "^1.4.4",
+    "express": "^4.17.1",
+    "fs": "0.0.1-security",
```
Comment (demns) [inline package.json]: where did this one came from?) it's inside node already

File: src/app.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+const fs = require('fs');
+const swaggerUi = require('swagger-ui-express');
+const YAML = require('yamljs');
+const swaggerDocument = YAML.load('./swagger.yaml');
+const { setCookie, deleteCookie, authorizeByCookie } = require('./cookie-handler.js');
+const postsController = require('./posts-controller.js');
+const commentsController = require('./comments-controller.js');
+const userController = require('./users-controller.js');
+
+var accessLogStream = fs.createWriteStream(path.join(__dirname, 'access.log'), { flags: 'a' });
```
Comment (demns) [inline src/app.js]: let

File: src/comments-controller.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+const commentsCollection = JSON.parse(fs.readFileSync(commentsDataLocation, 'utf8'));
+
+const getCommentsByPostId = function (req, res) {
+  res.json(commentsCollection.filter(comment => comment.postId == req.postId));
+};
+
+const createComment = function (req, res) {
+  const comment = req.body;
+  //TODO rewrite validation
+  if (!comment || !comment.summary || !comment.authorId) res.status(400).send('Bad request');
```
Comment (demns) [inline src/comments-controller.js]: not sure, but maybe you can ask swagger to do that check for you

File: src/cookie-handler.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
+const fs = require('fs');
+const bcrypt = require('bcrypt');
+const auth = require('./auth.js');
+
+var usersCollection = JSON.parse(fs.readFileSync('./data/users.json', 'utf8'));
+const cookieName = 'logged-in';
+
+//Login
+const setCookie = function (req, res) {
+    var cookie = req.cookies[cookieName];
```
Comment (demns) [inline src/cookie-handler.js]:
```suggestion
    let cookie = req.cookies[cookieName];
```

File: src/cookie-handler.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    if (cookie === undefined) {
```
Comment (demns) [inline src/cookie-handler.js]: !cookie? also you can just rewrite the whole if the other way round, that would reduce complexity

File: src/cookie-handler.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        if (!userFound) {
+            res.status(401).send('Wrong user, try /login again');
+        } else if (!checkPasswordForUser(userFound, password)) {
+            res.status(401).send('Wrong password, try /login again');
+        } else {
+            res.cookie(cookieName, login);
+            res.status(200).send('Logged in');
+        }
+    } else {
+        res.status(200).send('Already logged in, log out first');
```
Comment (demns) [inline src/cookie-handler.js]: I don't care but it's not necessarily 200

**Comment by demns** [comment]
короче всё круто с первого взгляда

File: src/cookie-handler.js (hunk 6 above)
Comment (polina-moisenia) [reply src/cookie-handler.js]: 30x redirect?

Comment (demns) [reply src/cookie-handler.js]: yes

**Pull request merged**